### ISSUE TITLE
observer_elevation error

### ISSUE THREAD

**Issue by flyize** [pnbruckner/ha-sun2#130]
The docs say this is optional, but I see this error every time it starts. Did I miss something?

```
This error originated from a custom integration.

Logger: custom_components.sun2.config
Source: custom_components/sun2/config.py:224
integration: Sun2 (documentation, issues)
First occurred: 9:08:10 AM (1 occurrences)
Last logged: 9:08:10 AM

New config option observer_elevation missing @ data[sun2][0], will use system general elevation setting
```

**Comment by pnbruckner** [comment]
@flyize, what version of this integration are you using, and what version of HA?

Are you using UI or YAML config? Please provide details of the configuration that leads to this error.

**Comment by flyize** [comment]
Looks like I'm using 3.3.3. And 2024.8.2 of HA.

I'm using the UI config, but do have one configured via YAML.

**Comment by pnbruckner** [comment]
I believe the warning stems from your YAML configuration. It would be nice if you could provide a copy of that configuration as requested, but I guess that's probably not necessary after all.

This is probably a documentation issue. `observer_elevation` is optional, but when not provided, the default is not zero, but rather the system's elevation configuration is used. This was done so that existing installations wouldn't change when upgrading. However, a warning is issued because using the system's elevation configuration is not recommended. The problem is, that setting is defined as elevation above _sea_ level, where the sun2 integration (and the built-in sun integration, for that matter) wants elevation above _ground_ level. This has actually been an issue in HA for years that has never been noticed or fixed.

So, yes, `observer_elevation` is optional, but not providing it results in a warning (not an error) so that you are aware that the integration may be getting the wrong value. If your system's elevation configuration is set to distance above _ground` level, then the sun2 integration will work correctly. If not, then it's probably best to add `observer_elevation` to your sun2 YAML configuration and set it to the appropriate value above ground level (which will also cause the warning to go away.)

I'll leave this open for now to remind me to update the docs. Feel free to comment further if you disagree or some of that doesn't make sense.

**Comment by flyize** [comment]
Oh! That makes much more sense. Thank you!

**Comment by pnbruckner** [comment]
README.md updated accordingly.

**Comment by zavjah** [comment]
Hi there, I found this warning in my log, too, and I tried to set up the observation_elevation. As I didn't find any other way I deleted the main entry and created a new one without using the HA name and location. Like this I was able to set the observer_level. 

Unfortunately, the warning was again the log after the restart because the integration created a new location by itself using the HA name and location, although I have manually used the same name. SO now there are two locations with the same name:

![Image](https://github.com/user-attachments/assets/bf07d0fc-33dc-442b-9012-e28adfdb5695)

Why is this?

**Comment by pnbruckner** [comment]
@zavjah, you probably have a YAML entry you need to delete. (Only the UI created entries have the CONFIGURE option.)

**Comment by zavjah** [comment]
@pnbruckner : Yes, you are right. This was in my configuration.yaml and once I removed it warning was gone and second location was not generated:

sun2:
  - unique_id: home

thx a lot
